test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/IdentityHashMapTest.scala

```diff
@@ -0,0 +1,397 @@
+
+    mp.put(testObj(4), null)
+    assertTrue(mp.containsKey(testObj(4)))
+    assertFalse(mp.containsKey(TestObj(4)))
+    assertTrue(keySet.contains(testObj(4)))
+    assertFalse(keySet.contains(TestObj(4)))
+    assertFalse(keySet.remove(TestObj(4)))
+    assertFalse(mp.containsKey(TestObj(4)))
+    assertTrue(mp.containsKey(testObj(4)))
+    assertTrue(keySet.contains(testObj(4)))
+    assertTrue(keySet.remove(testObj(4)))
+    assertFalse(mp.containsKey(testObj(4)))
+    assertFalse(keySet.contains(testObj(4)))
+  }
+
+  @Test def testIdentityValuesIsViewForQueriesWithCustomObjects(): Unit = {
+    val mp = factory.empty[TestObj, TestObj]
+    mp.put(testObj(1), testObj(11))
+    mp.put(testObj(2), testObj(22))
+    val values = mp.values()
+
+    assertTrue(values.contains(testObj(11)))
+    assertFalse(values.contains(TestObj(11)))
+    assertTrue(values.contains(testObj(22)))
+    assertFalse(values.contains(testObj(33)))
+
+    mp.put(testObj(3), testObj(33))
+
+    assertTrue(values.contains(testObj(33)))
+
+    val coll1 = ju.Arrays.asList(testObj(11), TestObj(22), testObj(33))
+    assertFalse(values.containsAll(coll1))
+
+    val coll2 = ju.Arrays.asList(testObj(11), testObj(22), testObj(33), testObj(44))
+    assertFalse(values.containsAll(coll2))
+  }
+
+  @Test def testIdentityValuesIsViewForRemoveWithCustomObjects(): Unit = {
+    assumeFalseIdentityHashMapViews()
+
+    val mp = factory.empty[TestObj, TestObj]
+    mp.put(testObj(1), testObj(11))
+    mp.put(testObj(2), testObj(22))
+    mp.put(testObj(3), testObj(33))
+    val values = mp.values()
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+
+    assertTrue(values.remove(testObj(11)))
+    assertTrue(values.remove(testObj(22)))
+    assertFalse(values.remove(TestObj(33)))
+
+    assertFalse(mp.containsKey(testObj(1)))
+    assertFalse(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+
+    mp.put(testObj(1), testObj(11))
+    mp.put(testObj(2), testObj(22))
+    mp.put(testObj(3), testObj(33))
+
+    val coll1 = ju.Arrays.asList(testObj(11), testObj(22), TestObj(33))
+    values.removeAll(coll1)
+
+    assertFalse(mp.containsKey(testObj(1)))
+    assertFalse(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3))) // fails JVM
+
+    mp.put(testObj(1), testObj(11))
+    mp.put(testObj(2), testObj(22))
+    mp.put(testObj(3), testObj(33))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+
+    values.retainAll(ju.Arrays.asList(testObj(11), testObj(22), TestObj(33)))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertFalse(mp.containsKey(testObj(3))) // fails JVM
+  }
+
+  // tests ported from Harmony
+  // tests with null keys and values
+  // tests share the same map so order matters
```

Comment:
Please remove the tests ported from Harmony. They do not add any value compared to the tests coming from `MapTest`, and the fact that they rely on order is absolutely terrible, since JUnit has options to execute tests in different orders (e.g., alphabetic).

Reply:
Done, the first time in Scala Native these were helpful but given the fine tests in Scala.js, they are superfluous.